buf/validate/internal/extra_func.cc

```diff
@@ -207,6 +207,9 @@ bool IsPort(const std::string_view str) {
   if (str.empty()) {
     return false;
   }
+  if (str.length() > 1 && str[0] == '0') {
+    return false;
+  }
```

Comment:
Same as https://github.com/bufbuild/protovalidate-go/pull/220: Consistently reject leading zero in port number in `isHostAndPort` - e.g. "080" and "00".